### PR TITLE
Update STM32 HAL with our new coding standard

### PR DESCRIPTION
**Pull request by Jeff-Ciesielski**
This is about 90% a cosmetic change.  There were two minor code
changes (in usart and can) where I added unrolling goto statements to
remove 'tricky' expressions in init code to replace very large chunks of
logically ORd together variables.  I also added some grouping parens
around logical comparisons to avoid confusion.

Code builds and the test suite passes as expected.

Signed-off-by: Jeff Ciesielski <jeffciesielski@gmail.com>

### CODE REVIEW

File: stm32_base/hal/ADC_stm32/ADC_device_stm32.c [hunk 1]
```diff
@@ -22,15 +22,15 @@
 #define ADC_PORT_VOLTAGE_RANGE 		5.0f
 #define ADC_SYSTEM_VOLTAGE_RANGE	20.0f
 
-#define SCALING_5V 					0.00125691302f
-#define SCALING_BATTERYV			0.00465f
+#define SCALING_5V 		0.00125691302f
```
Comment (stieg) [inline stm32_base/hal/ADC_stm32/ADC_device_stm32.c]: What rule on alignments are you following exactly?

File: stm32_base/hal/ADC_stm32/ADC_device_stm32.c [hunk 2]
```diff
@@ -76,15 +80,15 @@ int ADC_device_init(void) {
...
 	RCC_AHB1PeriphClockCmd(RCC_AHB1Periph_DMA2, ENABLE);
 	RCC_APB2PeriphClockCmd(RCC_APB2Periph_ADC2, ENABLE);
 
-	/* DMA2_Stream0 channel0 configuration **************************************/
+	/* DMA2_Stream0 channel0 configuration ************************************* */
 	DMA_DeInit(DMA2_Stream2);
 	DMA_InitStructure.DMA_Channel = DMA_Channel_1;
 	DMA_InitStructure.DMA_PeripheralBaseAddr = (uint32_t) & ADC2->DR;
-	DMA_InitStructure.DMA_Memory0BaseAddr = (uint32_t) &ADCConvertedValues[0];
+	DMA_InitStructure.DMA_Memory0BaseAddr = (uint32_t) & ADCConvertedValues[0];
```
Comment (stieg) [inline stm32_base/hal/ADC_stm32/ADC_device_stm32.c]: Does the linux standard call for space between & character and variable?

File: stm32_base/hal/CAN_stm32/CAN_device_stm32.c [hunk 3]
```diff
@@ -32,34 +32,50 @@ static xQueueHandle xCan2Rx = NULL;
...
 		xCan1Rx = xQueueCreate(CAN_QUEUE_LENGTH,
-				( unsigned portBASE_TYPE ) sizeof( CanRxMsg ));
+				       (unsigned portBASE_TYPE)
+				       sizeof(CanRxMsg));
 		xCan1Tx = xQueueCreate(CAN_QUEUE_LENGTH + 1,
-				( unsigned portBASE_TYPE ) sizeof( CanRxMsg ));
+				       (unsigned portBASE_TYPE)sizeof(CanRxMsg));
+		if (xCan1Rx == NULL || xCan1Rx == NULL) {
+			success = 0;
+			goto cleanup_and_return;
```
Comment (stieg) [inline stm32_base/hal/CAN_stm32/CAN_device_stm32.c]: Goto is find here, but why not simply return 0 here?  Seems like goto is unnecessary.

**Comment by stieg** [comment]
Only minor nits really.  Thanks for handling some of the cleanup.

File: stm32_base/hal/ADC_stm32/ADC_device_stm32.c (hunk 2 above)
Comment (Jeff-Ciesielski) [inline stm32_base/hal/ADC_stm32/ADC_device_stm32.c]: That's an accident, Will fix.

File: stm32_base/hal/ADC_stm32/ADC_device_stm32.c (hunk 1 above)
Comment (Jeff-Ciesielski) [inline stm32_base/hal/ADC_stm32/ADC_device_stm32.c]: Even alignment in blocks of definitions.  No hard and fast rule, just that a block of definitions with varying length have aligned constants.

File: stm32_base/hal/CAN_stm32/CAN_device_stm32.c (hunk 3 above)
Comment (Jeff-Ciesielski) [inline stm32_base/hal/CAN_stm32/CAN_device_stm32.c]: That could have been done as well.  Tradeoff between multiple return paths and a single exit point.

**Pull request merged**